### DISCUSSION TITLE
Smart Meter - How to ignore zero values?

### DISCUSSION THREAD

**Discussion by matoxp** [arendst/Tasmota#15511] [Q&A] [answered]
I'm using the Tasmota Smart Meter Interface for interfacing with a power meter (via OBIS ASCII) that reports monotonically increasing energy consumption (KwH). I noticed that sometimes  (once or twice a week) the value read from the meter is zero for a single sample. On next read the correct value is returned again (e.g. 5234, **0**, 5235). This is causing some issues with home-assistant's energy dashboard (as it interprets this as a counter-reset/rollover).

Is there a way to "preprocess" read values from the meter in Tasmota before emitting them via MQTT? E.g. if the value is zero or lower than the previous sample, either don't transmit the reading or use the previous value instead.

**Comment by FransOv** [comment]
Can you use the filter sensor in Home Assistant to remove the outliers or is this not possible with the energy dashboard.

**Reply by ZsZs73** [reply]
The median filter does not prevent the script to send zero value right after the boot of tasmota (in case the script starts in the middle of the meter transmission)

**Reply by gemu2015** [reply]
check the docs for smlj
this should solve your issue

**Comment by gemu2015** [comment, marked as answer]
check the docs and search for median filter

https://tasmota.github.io/docs/Smart-Meter-Interface/

**Reply by matoxp** [reply]
sml_median is not ideal for my use case - e.g. when processing the sample values 100,100,0,100,102 it will pick 100. The outlier filter in home-assistant does a slightly better job (as long as the 0 sample is not last in the list) as it also lets you specify a radius.

**Reply by helander** [reply]
As far as I can see no crc:s are calculated and therefore not checked. This is probably why you get wrong values every now and then. There is really no other cure to this than inclusion of crc calculations/checks in the code.

**Comment by helander** [comment]
As far as I can see no crc:s are calculated and therefore not checked. This is probably why you get wrong values every now and then. There is really no other cure to this than inclusion of crc calculations/checks in the code.

**Comment by hohwille** [comment]
Isnt it called *smart* home?
A energy sensor providing values in kWH is typically know to send monotonously growing values by design.
If you are at a value 42293.3kWH and then suddenly receive a value of 0kWH but then as a next value 42293.4kWH it does not seem smart that Home Assistant is aggregating a energy consumption of -42293.3kWH and then for the next slot of +42293.4kWH.
![grafik](https://github.com/user-attachments/assets/9957d901-912d-4ef0-9ba8-749de3c67ca1)
![grafik](https://github.com/user-attachments/assets/17ee7ee0-0dd5-4370-afb2-c6019d826275)

HomeAssisant would be smart if it sanatizes the values and consider 0kWH as a faulty measurement to be ignored.
Tasmota would be smart to know that the value is initially 0 on boostrapping and only after a positive value has been received, it should start sending data via MQTT. 

You may argue that this is a problem of the sensor and could be solved by Tasmota but that is also not working - see https://github.com/arendst/Tasmota/discussions/21126
